### DISCUSSION TITLE
Night exposure period issue

### DISCUSSION THREAD

**Discussion by czimm74** [aaronwmorris/indi-allsky#778] [General]
Been letting the camera run along and play nicely.. Sky on the other hand has been a bully (Canadian smoke most times).

Anyway, I have noticed that while I have the night exposure set to 20 seconds max, and the delay between exposures at 15 seconds.  The actual time between exposures varies, it looks like at the beginning and end of the night cycle the exposures seem to be 15 seconds apart but during the night they are on average 2 minutes apart.

I have been trying all kinds of variations on the two times with no apparent difference in timing.

**Comment by aaronwmorris** [comment]
Remind me of the type of camera you are using?

**Comment by czimm74** [comment]
IMX477
Arducam 12.3MP HQ

**Reply by aaronwmorris** [reply]
I am assuming you are using JPEG mode and not RAW mode?

**Reply by czimm74** [reply]
Yes, I can switch over to raw and see if anything different happens.  Have not thought about switching that

**Reply by aaronwmorris** [reply]
One of the key differences between RAW and JPEG mode (other than RAW and JPEG :-) ) is the JPEG mode utilizes the Auto-White Balance feature of libcamera.

AWB works very well when there is sufficient illumination, but it does not work very well at night.  libcamera will continuously take exposures until it can find a proper white balance... this can take 1-2 minutes for every exposure, regardless of the configured exposure time.  You still get the proper exposure, but whenever libcamera decides to give it to you.

**Reply by czimm74** [reply]
Ok, switched over to raw mode.  Will see how it runs tonight.  Didn't think that the difference between raw and jpg would cause that much interference, interesting.

**Reply by aaronwmorris** [reply]
I could disable the AWB in JPEG mode, just as you can enable it in RAW mode.  I just enabled it in JPEG mode because people kept complaining about the color balance.

Merged #779 so that the AWB mode is configured independently from the image mode.
